pkg/registry/manifest.go

```diff
@@ -30,76 +30,93 @@ func (c *Client) Manifest(image Image, dbManifest Manifest) (Manifest, error) {
 	ctx, cancel := c.timeoutContext()
 	defer cancel()
 
-	if c.sysCtx.DockerAuthConfig == nil {
-		c.sysCtx.DockerAuthConfig = &types.DockerAuthConfig{}
-		// TODO: Seek credentials
-		//auth, err := config.GetCredentials(c.sysCtx, reference.Domain(ref.DockerReference()))
-		//if err != nil {
-		//	return nil, errors.Wrap(err, "Cannot get registry credentials")
-		//}
-		//*c.sysCtx.DockerAuthConfig = auth
-	}
-
-	imgRef, err := ParseReference(image.String())
+	rmRef, err := ParseReference(image.String())
 	if err != nil {
 		return Manifest{}, errors.Wrap(err, "Cannot parse reference")
 	}
 
-	var imgDigest digest.Digest
-	if c.opts.CompareDigest {
-		imgDigest, err = docker.GetDigest(ctx, c.sysCtx, imgRef)
-		if err != nil {
-			return Manifest{}, errors.Wrap(err, "Cannot get image digest from HEAD request")
-		}
+	// Retrieve remote digest through HEAD request
+	rmDigest, err := docker.GetDigest(ctx, c.sysCtx, rmRef)
+	if err != nil {
+		return Manifest{}, errors.Wrap(err, "Cannot get image digest from HEAD request")
+	}
 
-		if dbManifest.Digest != "" && dbManifest.Digest == imgDigest {
-			return dbManifest, nil
-		}
+	// Digest match, returns db manifest
+	if c.opts.CompareDigest && len(dbManifest.Digest) > 0 && dbManifest.Digest == rmDigest {
+		return dbManifest, nil
 	}
 
-	imgCloser, err := imgRef.NewImage(ctx, c.sysCtx)
+	rmCloser, err := rmRef.NewImage(ctx, c.sysCtx)
 	if err != nil {
 		return Manifest{}, errors.Wrap(err, "Cannot create image closer")
 	}
-	defer imgCloser.Close()
+	defer rmCloser.Close()
 
-	rawManifest, _, err := imgCloser.Manifest(ctx)
+	rmRawManifest, _, err := rmCloser.Manifest(ctx)
 	if err != nil {
 		return Manifest{}, errors.Wrap(err, "Cannot get raw manifest")
 	}
 
-	if !c.opts.CompareDigest {
-		imgDigest, err = manifest.Digest(rawManifest)
+	// For manifests list compare also digest matching the platform
+	rmManifestMimeType := manifest.GuessMIMEType(rmRawManifest)
```

Comment:
```go
	rmRawManifest, rmManifestMimeType, err := rmCloser.Manifest(ctx)
```
should be slightly more accurate and efficient.

Reply:
Yes thanks